src/store/flow/resource.ts

```diff
@@ -143,15 +143,17 @@ export function findResourceVariantOverModesWith(
 export function findResourceVariantOverModesOn(
     resource: IResourceDefinition,
     filter: IResourceDefinitionVariantOverModesFilter) {
-
   const
       keysForComparison = without(Object.keys(filter), 'modes'), // b/c we do explicit partial matching on modes
-      filterWithComparatorKeys = pick(filter, keysForComparison),
-      variant = find<IResourceDefinitionVariantOverModes>(
-          resource.values,
-          v => isEqual(filterWithComparatorKeys, pick(v, keysForComparison))
-              && difference(filter.modes, v.modes).length === 0)
-
+      filterWithComparatorKeys = pick(filter, keysForComparison) //eg: {"languageId":"1","contentType":"text"}
+  let filterWithComparatorKeysAndArrayContentType = filterWithComparatorKeys //eg: {"languageId":"1","contentType":["text"]} //TODO: we may need to remove this one, once we have a clear idea which schema is used
+  if(!Array.isArray(filterWithComparatorKeysAndArrayContentType.contentType)) {
+    // @ts-ignore
+    filterWithComparatorKeysAndArrayContentType.contentType = [filterWithComparatorKeysAndArrayContentType.contentType]
+  }
+  let variant = find<IResourceDefinitionVariantOverModes>(
+          resource.values, v => (isEqual(filterWithComparatorKeys, pick(v, keysForComparison)) || isEqual(filterWithComparatorKeysAndArrayContentType, pick(v, keysForComparison)))
```

Comment:
If I understand this correctly, could we just use filterWithComparatorKeysAndArrayContentType? Aren't they always going to be equal?

Reply:
@jcbashdown , thanks for pointing this out. You know what? I think this is a `block resource` fix attempt I made too. And as discussed in previous comment (https://github.com/FLOIP/flow-builder/pull/12#discussion_r501989526), let's avoid such fix in this current PR. 
I'll revert to the original state.

Reply:
reverted !